games/openrct2/Portfile

```diff
@@ -43,9 +73,42 @@ depends_lib-append  port:libsdl2 \
                     port:libzip \
                     port:curl \
                     port:freetype \
-                    port:libiconv
+                    port:libiconv \
+                    port:nlohmann-json
+
+compiler.cxx_standard       2017
+configure.cxxflags-append   -std=c++17
+
+post-extract {
+    file mkdir ${worksrcpath}/data/sequence
+    move {*}[glob -nocomplain ${workpath}/*.parkseq] \
+        ${worksrcpath}/data/sequence
+    file mkdir ${worksrcpath}/data/object
+    move ${workpath}/official {*}[glob -nocomplain ${workpath}/rct*] \
+        ${worksrcpath}/data/object
+}
+
+patch {
+    reinplace "s|git describe HEAD|echo v${version}|" CMakeLists.txt
+    reinplace "s|git rev-parse --abbrev-ref HEAD|echo master|" CMakeLists.txt
+    reinplace "s|git rev-parse --short HEAD|echo 0000000|" CMakeLists.txt
```

Comment:
If you reinplace, use an absolute path like ${worksrcpath}/CMakeLists.txt; don't assume the current working directory is valid.

I assume you're doing this because upstream assumes we're building in a git clone and we're not, and indeed the `git` command may not be available. Ideally upstream would fix their CMakeLists.txt to work even when building from a source tarball. Please add the URL to the upstream bug report about this as a comment here. If there isn't one yet, please file one.

Reply:
Regarding the object files, those aren't object files from the game. The original game is still required. The object files from openrct2 are metadata files that are used to load the original. IIRC these are JSON files that contain things like checksums.

If we don't add the other distfiles this way, the project will just download them during the configure phase (unless that is disabled).

Without doing this step, it becomes very difficult for a user to fix their installation to actually work, because the metadata files are required.